src/libs/elektra/keyset.c

```diff
@@ -1269,7 +1269,9 @@ ssize_t ksCopyInternal (KeySet * ks, size_t to, size_t from)
 }
 
 /**
- * Searches for the start and optionally end of the key hierachy rooted at @p root in @p ks.
+ * Searches for the start and optionally end of the key hierarchy rooted at @p root in @p ks.
+ * The hierarchy will only contain keys in the same namespace as @p root.
+ * If @p root is a cascading key, only cascading keys will be part of the hierarchy.
```

Comment:
```suggestion
```

Since we plan to disallow adding cascading keys to keysets, that sentence will just be a future inconsistency. The sentence above it should also already make it clear that this happens, e.g. `KEY_NS_CASCADING != KEY_NS_SYSTEM`.

Reply:
Yes and no. Other functions, like `ksBelow`, `ksCut`, ... treat cascading `root`s differently. This is also why I was confused what `ksFindHierarchy` does with cascading keys. I did not consider that `KEY_NS_CASCADING` would be treated like a "normal" namespace here.

Reply:
So even if we disallow cascading keys in keysets in the future, the documentation still needs to be updated, as `ksFindHierarchy` will not do anything with them.

Reply:
Well it's your decision. If you keep the note about cascading keys, please add a TODO (outside the doxygen comment) to remind us that this should be updated, when cascading keys are forbidden in keysets.